### PR TITLE
feat: Basic support for H160 and H256 accounts.

### PR DESCRIPTION
**Pull request by crystalin**
This is a basic support got H160 accounts (like in Moonbeam/Moonriver) substrate chains.
I suspect we could pass an argument to `validateAddressMiddleware` to decide which type to support instead of relying on the `0x` prefix, but I'm not too familiar with the project.

What is your suggestion ?

### CODE REVIEW

**Comment by TarikGul** [comment]
@crystalin Thank you for the PR! I will give it a thorough review and response come Monday.

File: src/middleware/validate/validateAddressMiddleware.ts
Comment (emostov) [inline src/middleware/validate/validateAddressMiddleware.ts:39]:
```suggestion
			return [false, `Invalid decoded address length ${decoded.length} for H160 account`];
```

**Comment by emostov** [comment]
We need to update tests and make sure to have both success and failure tests for standard substrate addresses and H160.

File: src/middleware/validate/validateAddressMiddleware.ts
Comment (emostov) [inline src/middleware/validate/validateAddressMiddleware.ts:35]: What if someone submits a normal substrate address in a raw hex form? I think we should double check that `base58Decode` would error on that. But I suspect it will validate the raw address form as well as the checksum, so if that is the case we should continue supporting those formats. If this is indeed the case I suggest we first try `base58Decode` and if that fails _then_ check for `0x` and the length of 20

Comment (crystalin) [reply src/middleware/validate/validateAddressMiddleware.ts]: I updated the logic, can you check ?

**Review by emostov — APPROVED** [review]
thanks for the contribution - LGTM!

File: src/middleware/validate/validateAddressMiddleware.ts
Comment (dvdplm) [inline src/middleware/validate/validateAddressMiddleware.ts:28]: Except now it does, for the `H160` case. :/

File: src/middleware/validate/validateAddressMiddleware.ts
Comment (dvdplm) [inline src/middleware/validate/validateAddressMiddleware.ts:36]: Could we use `base58Validate` here instead?

File: src/middleware/validate/validateAddressMiddleware.ts
Comment (dvdplm) [inline src/middleware/validate/validateAddressMiddleware.ts:44]: There is `isEthereumAddress` that seems to do this job (and also runs more checks with `isEthereumChecksum`). Does it not work here?

File: src/middleware/validate/validateAddressMiddleware.ts
Comment (dvdplm) [inline src/middleware/validate/validateAddressMiddleware.ts:40]: "decoded" here is a bit unclear. On line 36 the address is decoded from base58 to hex. Here though, "decoding" means a different thing: cast a hex string to bytes. I would prefer clearer naming to properly distinguish the kind of decoding happening. Maybe `hexAddress`?

**Review by dvdplm — CHANGES_REQUESTED** [review]
Seems like this could be done in a better way, but I could be wrong.

**Comment by crystalin** [comment]
@dvdplm , I tried to improve the logic, please have a look

File: src/middleware/validate/validateAddressMiddleware.ts
Comment (TarikGul) [inline src/middleware/validate/validateAddressMiddleware.ts:37]: 👍

**Comment by TarikGul** [comment]
LGTM, I will approve once the lint check is fixed. Thanks again for the PR.

NOTE: I tested this myself against a moonriver node, and everything worked as expected.

**Review by TarikGul — APPROVED** [review]
@crystalin This will make it into the next Sidecar release v8.1.0. @dvdplm Once you approve the changes made in response to your suggestions I will get this updated into the next Sidecar release for either tonight or tomorrow morning.

**Comment by TarikGul** [comment]
> This will make it into the next Sidecar release v8.1.0.

This was premature insight of mine.

So we actually had a quick conversation about moving forward with this PR. We are actually going to leave it out of the next release to give this a second audit, we believe there might be a better way to do this, but just need to double check some alternatives. 

Moving forward with the 8.0.1 release and once this is merged we can always do another release.

**Comment by crystalin** [comment]
@TarikGul it is fine for us now, we are hosting our custom version at the moment

**Review by dvdplm — APPROVED** [review]
lgtm, thank you @crystalin for humoring me. :)

**Comment by emostov** [comment]
What is the status of this?

**Comment by TarikGul** [comment]
@emostov Im happy to get this merged in.

**Comment by TarikGul** [comment]
My main concern and delay was wondering if there was a better alternative to accepting a set of addresses within our middleware, but I think for now this is acceptable.

**Comment by emostov** [comment]
@TarikGul - yeah I wouldn't say I have a full picture of all the different address schemes, but from what I understand this solution should additionally except ethereum addresses on top of what we already except. If for some reason we found out accepting both is not desirable I suppose we could add some config option. Not sure what other ways we could do this?

**Comment by TarikGul** [comment]
@emostov Yea I agree with everything you said. I am going to do some last minute sanity checks, and if everything checks out again (which it should), I merge this tonight, and have it in tomorrow's release

**Comment by TarikGul** [comment]
Merging.

**Pull request merged**
